src/main/java/homework_6/map_problems_generator/MapProblemsMutableGenerator.java

```diff
@@ -0,0 +1,38 @@
+package homework_6.map_problems_generator;
+
+import java.util.Random;
+
+public class MapProblemsMutableGenerator {
+
+    private int a;
+
+    public MapProblemsMutableGenerator(int a) {
+        this.a = a;
+    }
+
+    public int getA() {
+        return a;
+    }
+
+    public void setA(int a) {
+        this.a = a;
+    }
+
+    @Override
+    public String toString() {
+        return String.valueOf(a);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        MapProblemsMutableGenerator that = (MapProblemsMutableGenerator) o;
+        return a == that.a;
+    }
+
+    @Override
+    public int hashCode() {
+        return new Random().nextInt(Integer.MAX_VALUE-1);
```

Comment:
I agree it's unclear in the task in homework chat, but your hashcode should be valid. Even with a valid hashcode mutable key cause problems in Hashmap